pkg/framework/plugins/pluginbuilder/pluginbuilder.go

```diff
@@ -25,6 +36,7 @@ type PluginBuilderAndArgsInstance struct {
 	// Just an example instance of this PluginArg so we can avoid having
 	// to deal with reflect Types
 	PluginArgInstance runtime.Object
+	Balance           bool
```

Comment:
This needs to be detected automatically by checking whether a plugin implements a specific interface. Also, a plugin builder does not need to know about any extension points as its purpose is to build plugins.